python/cudf/cudf/core/column_accessor.py

```diff
@@ -372,8 +372,11 @@ def insert(
             new_values = self.columns[:loc] + (value,) + self.columns[loc:]
             self._data = dict(zip(new_keys, new_values))
         self._clear_cache(old_ncols, old_ncols + 1)
-        # The type(name) may no longer match the prior label_dtype
-        self.label_dtype = None
+        try:
+            pd.Index([*list(self.names), name], dtype=self.label_dtype)
```

Comment:
What are we really checking here? Is there really no cheaper way to verify this than attempting to trigger an exception?

Reply:
We are checking if a new being added column name type matches existing columns type, if It matches we preserve the type else drop the type to create `object` dtype when fetched. I initially gated this with pandas-compatibility mode to only perform this inefficient check in `cudf.pandas`, do you want me to revert that check?

Reply:
> object dtype when fetched

The impact here instead should be when converting the column labels `to_pandas_index` is that the `pandas.Index` will infer the type from the column labels, which won't be a pandas nullable type since they aren't the default.

Was there a specific test related to the pandas nullable extension types that failed without this change? I'd lean on punting on this level of support if so.

Reply:
This is the pytest that failed `tests/groupby/methods/test_value_counts.py::test_compound`

Instead of fully punting on this, shall we go back to letting this support be gated by pandas-compatibility mode?